packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -308,23 +306,37 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 					"Unexpected error: No Vite HTTP server"
 				);
 
-				miniflare = new Miniflare(
-					getDevMiniflareOptions(resolvedPluginConfig, viteDevServer)
-				);
+				if (miniflare) {
+					await miniflare.setOptions(
+						getDevMiniflareOptions(resolvedPluginConfig, viteDevServer)
+					);
+				} else {
+					miniflare = new Miniflare(
+						getDevMiniflareOptions(resolvedPluginConfig, viteDevServer)
+					);
+				}
 
 				await initRunners(resolvedPluginConfig, viteDevServer, miniflare);
-				const routerWorker = await getRouterWorker(miniflare);
 
 				const middleware = createMiddleware(
-					({ request }) => {
-						return routerWorker.fetch(toMiniflareRequest(request), {
-							redirect: "manual",
-						}) as any;
+					async ({ request }) => {
+						if (miniflare) {
```

Comment:
I don't think it should be possible for miniflare to be undefined here, right?
Should this be an assertion?

Reply:
Good point. That will mean I can remove the optionals in the WebSockets file too.

Reply:
[7133bf1](https://github.com/cloudflare/workers-sdk/pull/8652/commits/7133bf117de8714ea78a29b0b155628d54cf61f7)